airflow/providers/google/suite/operators/drive.py

```diff
@@ -0,0 +1,118 @@
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account
+    :return: Remote file ids after upload
+    :rtype: Sequence[str]
+    """
+
+    template_fields = (
+        'local_paths',
+        'drive_folder',
+    )
+
+    def __init__(
+        self,
+        local_paths: Union[Sequence[Path], Sequence[str]],
+        drive_folder: Union[Path, str],
+        gcp_conn_id: str = "google_cloud_default",
+        delete: bool = False,
+        chunk_size: int = 100 * 1024 * 1024,
+        resumable: bool = False,
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.local_paths = local_paths
+        self.drive_folder = drive_folder
+        self.gcp_conn_id = gcp_conn_id
+        self.delete = delete
+        self.chunk_size = chunk_size
+        self.resumable = resumable
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context: "Context") -> Sequence[str]:
+        hook = GoogleDriveHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+        remote_file_ids = []
+
+        for local_path in self.local_paths:
+            self.log.info(f'Uploading file to Google Drive: {local_path}')
+
+            try:
+                remote_file_id = hook.upload_file(
+                    local_location=str(local_path),
+                    remote_location=str(Path(self.drive_folder) / Path(local_path).name),
+                    chunk_size=self.chunk_size,
+                    resumable=self.resumable,
+                )
+
+                remote_file_ids.append(remote_file_id)
+
+                if self.delete:
+                    os.remove(local_path)
+                    self.log.info(f'Deleted local file: {local_path}')
+            except FileNotFoundError:
+                self.log.warning(f"File {local_path} can't be found")
+
+        self.xcom_push(context, "remote_file_ids", remote_file_ids)
+        return remote_file_ids
```

Comment:
This seems redundant. When the operator completes there will be 2 `XComs` with the same information but different keys: "return_value" and "remote_file_ids". This feels more straightforward but no strong opinion really on which one is used.
```suggestion
        return remote_file_ids
```